scala-generator/src/main/scala/models/Play2Json.scala

```diff
@@ -94,7 +104,15 @@ case class Play2Json(
               """(js \ "value").validate[String] match {""",
               Seq(
                 s"case play.api.libs.json.JsSuccess(v, _) => play.api.libs.json.JsSuccess(${enum.qualifiedName}(v))",
-                "case err: play.api.libs.json.JsError => err"
+                "case err: play.api.libs.json.JsError =>",
+                Seq(
+                  s"""(js \\ "${enum.originalName}").validate[String] match {""",
```

Comment:
have we confirmed that this case needs to be handled? are we using this somewhere?

Reply:
was the previous generated Reads able to handle this?

Reply:
This is happening for a union type without a discriminator.

This change in the `Reads` is to allow to read a a type from union type without a discriminator by using the specific `Reads`.

So to support, without discriminator:

```
trait User
case class RegisteredUser() extends User

val js = Json.toJson(RegisteredUser()) // { "registered_user": {...}}
js.as[RegisteredUser] 
```